core/chains/evm/headtracker/head_tracker.go

```diff
@@ -131,63 +176,68 @@ func (ht *headTracker) Start(ctx context.Context) error {
 }
 
 // Close stops HeadTracker service.
-func (ht *headTracker) Close() error {
+func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) Close() error {
 	return ht.StopOnce("HeadTracker", func() error {
 		close(ht.chStop)
 		ht.wgDone.Wait()
 		return ht.broadcastMB.Close()
 	})
 }
 
-func (ht *headTracker) Name() string {
+func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) Name() string {
 	return ht.log.Name()
 }
 
-func (ht *headTracker) HealthReport() map[string]error {
+func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) HealthReport() map[string]error {
 	report := map[string]error{
 		ht.Name(): ht.StartStopOnce.Healthy(),
 	}
 	maps.Copy(report, ht.headListener.HealthReport())
 	return report
 }
 
-func (ht *headTracker) Backfill(ctx context.Context, headWithChain *evmtypes.Head, depth uint) (err error) {
+func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) Backfill(ctx context.Context, headWithChain HTH, depth uint) (err error) {
 	if uint(headWithChain.ChainLength()) >= depth {
 		return nil
 	}
 
-	baseHeight := headWithChain.Number - int64(depth-1)
+	baseHeight := headWithChain.BlockNumber() - int64(depth-1)
 	if baseHeight < 0 {
 		baseHeight = 0
 	}
 
-	return ht.backfill(ctx, headWithChain.EarliestInChain(), baseHeight)
+	earliestHead, ok := headWithChain.EarliestHeadInChain().(HTH)
```

Comment:
Made use of Type casting to avoid the circular import issue. 

The other alternative could be to use extra interfaces to avoid type casting, but i felt that that would make the `Head` bloated.

Would appreciate any other suggestions on how to improve the cleanliness of code here!

Reply:
Can `backfill` just accept `head commontypes.Head[BLOCK_HASH]` instead of `HTH`?

Reply:
Thanks so much for the note @jmank88. Really missed that out there. Appreciate it :)